cfme/infrastructure/datastore.py

```diff
@@ -157,14 +157,16 @@ def _on_vms_page(self):
         )
 
     def delete_all_attached_vms(self):
-        sel.force_navigate('infrastructure_datastore', context=self._get_context())
+        if not self._on_detail_page():
+            sel.force_navigate('infrastructure_datastore', context=self._get_context())
```

Comment:
It will be nice to extract these two lines of logic to a separate method called `load_details`. That is what we do for other models and it will enable us to apply some mixins that rely on that method. Check the load_details in providers and virtual machines to see how it is done :)